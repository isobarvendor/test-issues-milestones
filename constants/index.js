let language = {
  en: {
    browserTitle: "Turn Up Your Rhythm",
    meta: [
      {
        hid: "Turn Up Your Rhythm",
        name: "Turn Up Your Rhythm",
        content: "Home page Coke - Turn Up Your Rhythm"
      }
    ],
    periodDescription: "",
    signInButton: "SIGN IN <BR/><small>to win prizes</small>",
    menu: {
      home: "Home",
      privacy: "Privacy",
      terms: "Terms",
      logout: "Logout",
      myRewards: "My Rewards"
    },
    countdown: {
      days: "DAYS",
      hrs: "HRS",
      mins: "MINS",
      secs: "SECS"
    },
    submissionText: {
      header: "Enter your code",
      hello: "Hello",
      siginHeader:
        "Sign in or register a Coca-Cola <br>account to enter the draw",
      name: "Name",
      email: "Email",
      phoneNumber: "Phone Number",
      acceptTerm:
        'I accept the <a href="/tnc">Terms and Conditions</a> of this this redemption.',
      acceptPrivacy:
        'I accept the <a href="/privacy">Privacy Policy</a> of this redemption.',
      declareAge: "I declare that I am above 13  years old.",
      enterCode: "Enter Unique Code",
      tooltipText:
        "Enter the code found under the cap/tab of your Coca Cola purchase here",
      phoneTooltip:
        "Please input the correct phone number so that Coca-Cola can contact you directly incase you are the winner",
      buttonText: "Collect your prize",
      errorTerm: "Please accept our terms and conditions",
      errorPolicy: "Please accept our privacy policies",
      errorDeclare: "Please declare that you are above 13  years old",
      errorPinCode: "Oops your pin code is invalid or already redeemed",
      errorPinCode1: "Oops your pin code is invalid or already redeemed",
      errorPinCode2: "Oops your pin code is invalid or already redeemed",
      errorPinCode3: "Oops your pin code is invalid or already redeemed",
      errorPinCode4: "Oops your pin code is invalid or already redeemed",
      errorZaloPrize: "Redemption Limit for zalo prize has been reached",
      errorNormalPrize: "Redemption Limit for normal prize has been reached",
      errorAPI: "Oops something went wrong please try again",
      errorMaxPhone: "You reach maximum phone number length",
      errorRequiredPhone: "The phone number is required",
      participateAgain: "Participate again",
      listenNow: "Listen now",
      redeemPrize: "Redeem Prize",
      continue: "Continue with",
      note:
        "Please call 1800 555 584 (VND 1500/min, from 9:00 to 17:00, weekdays) to reveive redemption instructions",
      startRedeeming: "Start Redeeming",
      redeemNextPrize: "Redeem next prize",
      textPage:
        "You have entered the Entry Code from Coca-Cola 300ml PET bottle or Coca-Cola 235ml can, you will have 2 lucky draw opportunities",
      hardLuckHeader: "",
      hardLuckTitle:
        "Bạn chưa may mắn lần này. Nhưng vẫn còn nhiều quà hấp dẫn đang chờ. Hãy tiếp tục khui Coke và tham gia bạn nhé!",
      hardLuckImage: "/img/landing/hardluck.png"
    },
    winnerText: {
      header: "WINNER LIST",
      nowinner: "Coming soon"
    },
    globalError: {
      errorAPI: "Oops something went wrong",
      tryagain: "Please try again"
    },
    footer: {
      privacy: "Privacy Policy",
      terms: "Terms and conditions"
    }
  },
  kh: {
    browserTitle: "បញ្ចេញភាពរស់រវើករបស់អ្នក",
    meta: [
      {
        hid: "បញ្ចេញភាពរស់រវើករបស់អ្នក",
        name: "បញ្ចេញភាពរស់រវើករបស់អ្នក",
        content: "Home page Coke - បញ្ចេញភាពរស់រវើករបស់អ្នក"
      }
    ],
    periodDescription:
      "ចាប់ពីថ្ងៃទី 04 កញ្ញា 2021 ដល់ថ្ងៃទី 30 តុលា 2021 គ្រប់ផលិតផលកូកា-កូឡា អ្នកអាចមានឱកាសស្តាប់ចម្រៀងថ្មីៗ ពី Pleng (Unbox Music Playlist) និងរង្វាន់ជាច្រើនទៀត ដោយគ្រាន់តែអនុវត្តតាមជំហានដូចខាងក្រោម",
    signInButton: "ចុះឈ្មោះ<BR/><small>ដើម្បីមានឱកាសឈ្នះ</small>",
    menu: {
      home: "ទំព័រដើម",
      privacy: "គោលការណ៍ឯកជន",
      terms: "លក្ខន្ដិកនៃការចូលរួម",
      logout: "Logout",
      myRewards: "My Rewards"
    },
    countdown: {
      days: "DAYS",
      hrs: "HRS",
      mins: "MINS",
      secs: "SECS"
    },
    submissionText: {
      header: "វាយបញ្ចូលលេខកូដ",
      hello: "Hello",
      siginHeader: " ចុះឈ្មោះដើម្បីចូលរួម",
      name: "ឈ្មោះ",
      email: "អុីម៉េល",
      phoneNumber: "លេខទូរស័ព្ទ",
      acceptTerm: "ខ្ញុំយល់ព្រមទទួលយកនូវលក្ខន្ដិកនេះ",
      acceptPrivacy: "ខ្ញុំយល់ព្រមទទួលយកនូវគោលការណ៍ឯកជនភាពនេះ",
      declareAge: "ខ្ញុំសូមប្រកាសថាខ្ញុំមានអាយុលើស 13 ឆ្នាំ",
      enterCode: "វាយបញ្ចូលលេខកូដ",
      tooltipText:
        "វាយបញ្ចូលលេខកូដដែលឃើញពីក្រោមគម្របដប ឬក្រវិលកំប៉ុងកូកា-កូឡារបស់អ្នកនៅទីនេះ",
      phoneTooltip:
        "សូមដាក់លេខទូរស័ព្ទពិតប្រាកដរបស់អ្នកដើម្បីក្រុមការងារកូកា-កូឡាអាចទាក់ទងទៅលោកអ្នកនៅពេលដែលឈ្នះបានទូរស័ព្ទ និង​កាសទូរស័ព្ទ",
      buttonText: "ទទួលយករង្វាន់របស់អ្នក",
      errorTerm: "សូមចុចយល់ព្រមទទួលយកនូវលក្ខន្ដិកៈនៃការចូលរួមកម្មវិធី",
      errorPolicy: "សូមចុចយល់ព្រមទទួលគោលការណ៍ឯកជនភាព",
      errorDeclare: "សូមប្រកាសថាអ្នកមានអាយុលើពី ១៣ ឆ្នាំ",
      errorPinCode: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorPinCode1: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorPinCode2: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorPinCode3: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorPinCode4: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorZaloPrize: "Redemption Limit for zalo prize has been reached",
      errorNormalPrize: "Redemption Limit for normal prize has been reached",
      errorAPI: " បញ្ហាបច្ចេទេស សូមព្យាយាមម្តងទៀត",
      errorMaxPhone: "អ្នកបញ្ជូលលេខទូរស័ព្ទគ្រប់ខ្ទង់",
      errorRequiredPhone: " តម្រូវឲ្យមានលេខទូរស័ព្ទ",
      participateAgain: " ចូលរួមម្តងទៀត",
      listenNow: "ស្តាប់ឥឡូវនេះ",
      redeemPrize: " ប្តូរយករង្វាន់",
      continue: " ចុចបន្ត",
      note: "សូមទាក់ទងលេខ 1-800-20-2222 ដើម្បីដឹងពីរបៀបប្តូរយករង្វាន់",
      startRedeeming: "ចាប់ផ្តើមប្តូរយករង្វាន់",
      redeemNextPrize: "ប្តូរយករង្វាន់បន្ទាប់",
      thankyouMessage: "<h2>អបអរសាទរ! អ្នកឈ្នះរង្វាន់ទ្វេរដង!<h2/>",
      textPage:
        "អ្នកបានបញ្ចូលលេខកូដពីដបកូកា-កូឡាដបជ័រ 350ml អ្នកនឹងមានឱកាសចាប់រង្វាន់ 2 ដង",
      hardLuckHeader: "មានបញ្ហាបច្ចេទេស",
      hardLuckTitle: "",
      hardLuckImage: "",
      theCodeFieldIsRequired: "សូមបំពេញលេខកូដទីនេះ"
    },
    winnerText: {
      header: "តារាងឈ្មោះអ្នកឈ្នះ",
      nowinner: "មកដល់ឆាប់ៗ នេះ"
    },
    globalError: {
      errorAPI: "មានបញ្ហាបច្ចេទេស",
      tryagain: " សូមព្យាយាមម្តងទៀត"
    },
    footer: {
      privacy: "គោលការណ៍ឯកជន",
      terms: "លក្ខន្ដិកនៃការចូលរួម"
    }
  }
  // vn:{
  //     browserTitle:"Turn Up Your Rhythm",
  //     meta: [
  //       {
  //         hid: "Turn Up Your Rhythm",
  //         name: "Turn Up Your Rhythm",
  //         content: "Home page Coke - Turn Up Your Rhythm"
  //       }
  //     ],
  //     periodDescription:"Rất tiếc thời gian CTKM Bừng Giai Điệu Hứng Khởi đã kết thúc vào ngày 06/06/2021",
  //     signInButton:"ĐĂNG NHẬP <BR/><small>để trúng thưởng</small>",
  //     menu:{
  //       home:"Trang Chủ",
  //       privacy:"Điều khoản bảo mật",
  //       terms:"Điều kiện & Điều khoản áp dụng",
  //       logout:"đăng xuất",
  //       myRewards:"Giải thưởng của bạn"
  //     },
  //     countdown:{
  //       days:"DAYS",
  //       hrs:"HRS",
  //       mins:"MINS",
  //       secs:"SECS",
  //     },
  //     submissionText:{
  //       header:"",
  //       hello:"Hello",
  //       siginHeader:'Đăng kí và bắt đầu tham gia chương trình rút thăm trúng thưởng',
  //       name:"Họ & Tên",
  //       email:"Email",
  //       phoneNumber:"Số điện thoại",
  //       acceptTerm:'Tôi đồng ý với <a href="/tnc" target="_blank">Điều kiện và Điều khoản được áp dụng</a> cho chương trình khuyến mãi',
  //       acceptPrivacy:'Tôi đồng ý với <a href="/privacy" target="_blank">Điều khoản Bảo mật</a> của chương trình khuyến mãi',
  //       declareAge:'Tôi cam đoan đã đủ hoặc trên 18 tuổi',
  //       enterCode:'Nhập mã ngay',
  //       tooltipText:'Nhập mã dự thưởng bên dưới nắp khoen hay nắp chai của sản phẩm Coca-Cola có khuyến mãi',
  //       phoneTooltip:"Vui lòng nhập chính xác số điện thoại để BTC có thể liên lạc trong trường hợp bạn là người trúng giải",
  //       buttonText:'Nhận quà thưởng',
  //       errorTerm:'Vui lòng chấp nhận các điều khoản và điều kiện của chúng tôi',
  //       errorPolicy:"Vui lòng chấp nhận chính sách bảo mật của chúng tôi",
  //       errorDeclare:"Vui lòng tuyên bố rằng bạn trên 18 tuổi",
  //       errorPinCode:"Xin lỗi Mã Tham Dự này không đúng. Vui lòng kiểm tra lại.",
  //       errorPinCode1:"Xin lỗi Mã Tham Dự này không đúng. Vui lòng kiểm tra lại.",
  //       errorPinCode2:"Xin lỗi Mã Tham Dự này đã được sử dụng. Vui lòng kiểm tra lại.",
  //       errorPinCode3:"Xin lỗi Mã Tham Dự này không đúng. Vui lòng kiểm tra lại.",
  //       errorPinCode4:"Xin lỗi Mã Tham Dự này không đúng. Vui lòng kiểm tra lại.",
  //       errorZaloPrize:"Redemption Limit for zalo prize has been reached",
  //       errorNormalPrize:"Redemption Limit for normal prize has been reached",
  //       errorAPI:"Rất tiếc. Đã xảy ra lỗi. Vui lòng thử lại",
  //       errorMaxPhone:"Bạn đạt đến độ dài số điện thoại tối đa",
  //       participateAgain:"Participate again",
  //       listenNow:"Nghe nhạc ngay",
  //       redeemPrize:"ĐỔI GIẢI THƯỞNG TẠI ĐÂY",
  //       continue:"Tiếp tục với",
  //       note:"Vui lòng gọi hotline 1900555584 (1500VND/ phút từ 9:00 đến 17:00, ngày hành chính) để nhận hướng dẫn thủ tục nhận giải thưởng.",
  //       startRedeeming:"Bắt đầu rút thăm",
  //       redeemNextPrize:"Tiếp tục đổi giải",
  //       errorRequiredPhone:"The phone number is required",
  //       textPage:"Bạn vừa nhập Mã dự thưởng từ sản phẩm chai nhựa Coca-Cola 300ml hoặc Lon Coca-Cola 235ml. Bạn có 2 cơ hội rút thăm trúng thưởng",
  //       hardLuckHeader:"",
  //       hardLuckTitle:"Lượt nghe một bài nhạc độc quyền của Coca-Cola",
  //       hardLuckImage:"https://d1tnwdlrciourl.cloudfront.net/prizeImages/1e0b8kkwap0rj/kkwbxo87/ExclusiveSong.png",
  //       hardLuckMusic:"https://elasticbeanstalk-prod-appcontent-016176.s3-ap-southeast-1.amazonaws.com/can_mong_tay_master_11-4_fix.mp3",
  //     },
  //     winnerText:{
  //       header:"WINNER LIST",
  //       nowinner:"Coming soon"
  //     },
  //     globalError:{
  //       errorAPI:"Oops something went wrong",
  //       tryagain:"Please try again"
  //     },
  //     footer:{
  //       privacy:"Điều khoản bảo mật",
  //       terms:"Điều kiện & Điều khoản áp dụng"
  //     }

  //   }
};

export const translation = language[process.env.language];
