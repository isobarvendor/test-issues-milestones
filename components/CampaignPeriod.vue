<template>
  <div class="container campaign-period-section redbox-withwhiteborder ">
    <v-row no-gutters>

      <v-col
        cols="12"
        sm="12"
        md="4"
      >

          <v-row no-gutters class="logo-title">
               <v-col
                  cols="4"
                  md="12"
                  sm="4"
                >

                  <img src="/img/landing/COKE STUDIO.png" class="logo"  />
                </v-col>

                <v-col
                  cols="8"
                  md="12"
                  sm="8"
                  class="participate-text-container"
                >
                 <div class="tagline" v-html="howData.title"></div>
                </v-col>
          </v-row>

      </v-col>
      <v-col
        cols="12"
        sm="12"
        md="8"
        class="desc-container"
      >
          <div class="periode-container">
            <p>{{this.data.Title}}</p>
            <p class="campaign-period" v-html="startdate+' â€“ '+enddate"></p>
          </div>
          <div class="period-desc-container">
              {{periodDescription}}
          </div>

           <v-row no-gutters   >
               <v-col
                  cols="12"
                  md="4"
                  sm="12"
                   v-for="(item, idx) in howData.slides" :key="'steps'+idx"
                >
                    <v-row no-gutters  class="center-layout" >
                      <v-col
                          cols="5"
                          md="12"
                          sm="5"
                          class="participate-image-container"
                        >
                          <img v-if="$mq == 'sm'&&item.mobileImage.length>0" :src="item.mobileImage[0].url" class="participate-image" />
                          <img v-else-if="item.desktopImage.length>0&&$mq !== 'sm'" :src="item.desktopImage[0].url" class="participate-image" />
                          <img v-else :src="'/img/landing/drink.png'" />
                        </v-col>

                        <v-col
                          cols="7"
                          md="12"
                          sm="7"
                          class="participate-text-container"
                        >
                         <span v-html="item.description" />

                        </v-col>
                  </v-row>

                </v-col>

          </v-row>
            <div v-if="loginInfo" v-html="howData.description" class="description"></div>

      </v-col>
    </v-row>



  </div>
</template>

<script>
import moment from "moment";
import {translation} from "@/constants/index"
moment.locale('en', {
    ordinal: num => {
        const b = num % 10;
        const output = (~~(num % 100 / 10) === 1) ? 'th' :
            (b === 1) ? 'st' :
                (b === 2) ? 'nd' :
                    (b === 3) ? 'rd' : 'th';
        return num + '<sup>' + output + '</sup>';
    },
});

export default {
   name: "CampaignPeriod",
  data() {
    return {
      periodDescription:translation.periodDescription
    };
  },
  computed:{
    startdate (){
        return this.data && moment(this.data.fromDate).format("Do MMMM YYYY")
    },
    enddate (){
      return this.data && moment(this.data.toDate).format("Do MMMM YYYY");
    },
    loginInfo(){
       return this.$store.state.login;
    }


  }
  ,
  props: {
    data: null,
    howData:{}
  },
  beforeMount() {

  }
};
</script>

<style>
.campaign-period-section .description{
  text-align: center;
}
</style>
