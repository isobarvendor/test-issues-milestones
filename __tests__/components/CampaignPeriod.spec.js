

import { createLocalVue, shallowMount } from '@vue/test-utils'

import CampaignPeriod from "@/components/CampaignPeriod";
import {translation} from "@/constants/index"

import Vuex from 'vuex';


let localVue, store;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return [
            {Theme : {
              Theme : "Coke"
            },
            SectionMenu:{
              Menu:[
                {
                  Link:"Home"
                },
                {
                  Link:"Wallet"
                },
                {
                  Link:"Wallet"
                },
                {
                  Link:"Wallet"
                },
                {
                  Link:"Wallet"
                },
                {
                  Link:"Wallet"
                },
                {
                  Link:"Wallet"
                },
                {
                  Link:"Wallet"
                }
              ],
              Image:[
                {url :'image'}
              ],
              Description:"test description"

            }
          }
      ];
        }
      },
      state:{
        login:{
          name:"Ronald"
        }
      }
    });
  });

const factory = () => {
  return shallowMount(CampaignPeriod, {
    data() {
      return {
        periodDescription:translation.periodDescription
      };
    },
    localVue, store,
    propsData: {
      data :{
        fromDate:'2020-11-01',
        toDate:'2020-11-11',
        Title:"Campaign Period"
      },
      howData:{
        title:"test",
        slides:
        [
            {
              mobileImage:[{url:"mobile image"}],
              desktopImage:[{url:"desktop image"}],
              description:"tes"
            }
        ],
        description:"tes"
      }
    }
  });
};

describe("CampaignPeriod", () => {


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
