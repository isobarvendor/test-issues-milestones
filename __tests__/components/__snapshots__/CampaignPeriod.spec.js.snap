// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignPeriod renders properly 1`] = `
<div class="container campaign-period-section redbox-withwhiteborder ">
  <v-row-stub tag="div" nogutters="true">
    <v-col-stub cols="12" sm="12" md="4" tag="div">
      <v-row-stub tag="div" nogutters="true" class="logo-title">
        <v-col-stub cols="4" sm="4" md="12" tag="div"><img src="/img/landing/COKE STUDIO.png" class="logo"></v-col-stub>
        <v-col-stub cols="8" sm="8" md="12" tag="div" class="participate-text-container">
          <div class="tagline">test</div>
        </v-col-stub>
      </v-row-stub>
    </v-col-stub>
    <v-col-stub cols="12" sm="12" md="8" tag="div" class="desc-container">
      <div class="periode-container">
        <p>Campaign Period</p>
        <p class="campaign-period">1<sup>st</sup> November 2020 – 11<sup>th</sup> November 2020</p>
      </div>
      <div class="period-desc-container">
        กิจกรรมเปิดจังหวะซ่าในตัวคุณได้สิ้นสุดลงแล้ว ขอบคุณที่สนใจและเข้าร่วมกิจกรรมกับ โค้ก
      </div>
      <v-row-stub tag="div" nogutters="true">
        <v-col-stub cols="12" sm="12" md="4" tag="div">
          <v-row-stub tag="div" nogutters="true" class="center-layout">
            <v-col-stub cols="5" sm="5" md="12" tag="div" class="participate-image-container"><img src="desktop image" class="participate-image"></v-col-stub>
            <v-col-stub cols="7" sm="7" md="12" tag="div" class="participate-text-container"><span>tes</span></v-col-stub>
          </v-row-stub>
        </v-col-stub>
      </v-row-stub>
      <div class="description">tes</div>
    </v-col-stub>
  </v-row-stub>
</div>
`;
